guide/groovy.md

```diff
@@ -103,15 +94,26 @@ import org.commonjava.maven.ext.core.groovy.InvocationPoint
 @PMEBaseScript BaseScript pme
 ```
 
-where InvocationStage may be `FIRST`, `LAST` or `BOTH`. This denotes whether the script is ran
-before all other manipulators, after or both. The script therefore encodes how and when it is run.
-
 The API can then be invoked by e.g.
 
     pme.getBaseDir()
 
 <b>NOTE</b> : Be careful not to use <code>pme.getProperties()</code> or <code>pme.getProject().getProperties()</code> as that actually calls the [Groovy language API](http://docs.groovy-lang.org/latest/html/api/org/codehaus/groovy/runtime/DefaultGroovyMethods.html#getProperties(java.lang.Object))
 
+#### Invocation stages
+
+In the example script, we saw the use of the `@InvocationPoint` annotation which controls when the script is run. It
+takes a single argument, `invocationPoint`, with the type `InvocationStage`. The possible values for `InvocationStage`
+are `PREPARSE`, `FIRST`, `LAST`, and `ALL`. These values are relative to when the manipulations to the POM files are
+made. The table below provides a description of the invocation stages available for running a script.
+
+| Stage      | Description                                                                                                                                                                              |
+|------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `PREPARSE` | Runs the script _prior to parsing any POM files_ and before any manipulations have been performed, thus allowing the modification of POM files on disk before they are read into memory. |
+| `FIRST`    | Runs the script _first_ after all POM files have been read into memory, but before any manipulations have been performed.                                                                |
+| `LAST`     | Runs the script _last_ after all modifications to the in-memory POM files have been performed.                                                                                           |
+| `ALL`      | Runs the during _all_ possible stages: `PREPARSE`, `FIRST`, and `LAST`.  The `getInvocationStage()` API can be used to determine in which stage the script is currently running.         |
```

Comment:
I think we need to make it clearer the consequences of editing files on disk during FIRST/LAST i.e. the warning from above. Otherwise I like the new section

Reply:
Do you just want to move the warning section down here? I probably just deleted it and forgot to move it.

Reply:
I think it may make sense as a paragraph now instead of a straight warning since it is safe now in the `PREPARSE` phase, which is somewhat explained in the table, but could be explained more later.

Reply:
I think it needs more than just moving the section ; I agree you might need to reconsider how to structure it given the warning applies to some sections but not every one.